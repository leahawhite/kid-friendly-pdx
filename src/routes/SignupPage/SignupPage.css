.SignupPage {
  text-align: left;
  font-size: 22px;
  font-weight: bold;
  margin: 40px 20px 0 20px;
}