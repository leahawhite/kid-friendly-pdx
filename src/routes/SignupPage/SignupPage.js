import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import SignupForm from '../../components/SignupForm/SignupForm';
import './SignupPage.css';

export default class SignupPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }

  handleSignUpSuccess = () => {
    const { history } = this.props
    history.push('/login')
    // push validated email/pass as props to login page for form population (or auto-login?)
  }

  render() {
    return (
      <section className="SignupPage">
        <div className="SignupPage-container">
          <h2>Sign Up</h2>
          <SignupForm onSignUpSuccess={this.handleSignUpSuccess} />
          <div className="signup-reminder">Already have an account?{' '}
            <Link to='/login'>
              Log in.
            </Link>
          </div>
        </div>
      </section>
    )
  }
}