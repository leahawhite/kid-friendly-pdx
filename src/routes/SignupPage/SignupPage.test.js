import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignupPage from './SignupPage'

describe('SignupPage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <SignupPage />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<SignupPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
