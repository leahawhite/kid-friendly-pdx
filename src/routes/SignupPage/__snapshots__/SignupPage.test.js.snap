// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage component renders UI as expected 1`] = `
<section
  className="SignupPage"
>
  <div
    className="SignupPage-container"
  >
    <h2>
      Sign Up
    </h2>
    <SignupForm
      onSignUpSuccess={[Function]}
    />
    <div
      className="signup-reminder"
    >
      Already have an account?
       
      <Link
        to="/login"
      >
        Log in.
      </Link>
    </div>
  </div>
</section>
`;
